cmd/init.go

```diff
@@ -576,3 +577,42 @@ func parseProjectParm(projectParm string, config *RootCommandConfig) (string, st
 
 	return "", "", errors.New("malformed project parameter - something unusual happened")
 }
+
+func defaultProjectName(config *RootCommandConfig) string {
+	projectName, _ := getProjectName(config)
+
+	if projectName == "" || projectName == "my-project" {
+
+		projectDirPath, err := os.Getwd()
+
+		if err != nil {
+			Error.log(err)
+			os.Exit(1)
+		}
+
+		indexPathSeparator := strings.LastIndex(projectDirPath, string(os.PathSeparator))
+
+		projectName := projectDirPath[indexPathSeparator+1:]
+
```

Comment:
I believe that these two lines of code are unnecessary, because `filepath.Base` already gets you the leaf.